models/agent_helpers.go

```diff
@@ -44,9 +44,45 @@ func checkUniqueAgentID(q *reform.Querier, id string) error {
 	}
 }
 
-// FindAllAgents returns all Agents.
-func FindAllAgents(q *reform.Querier) ([]*Agent, error) {
-	structs, err := q.SelectAllFrom(AgentTable, "ORDER BY agent_id")
+// FindAgents returns Agents by filters.
+func FindAgents(q *reform.Querier, filters AgentFilters) ([]*Agent, error) {
+	var conditions []string
+	var args []interface{}
+	idx := 1
+	if filters.AgentType != nil {
+		conditions = append(conditions, fmt.Sprintf("agent_type = %s", q.Placeholder(idx)))
+		args = append(args, *filters.AgentType)
+		idx++
+	}
+	if filters.NodeID != "" {
+		if _, err := FindNodeByID(q, filters.NodeID); err != nil {
+			return nil, err
+		}
+		conditions = append(conditions, fmt.Sprintf("node_id = %s", q.Placeholder(idx)))
+		args = append(args, filters.NodeID)
+		idx++
+	}
+	if filters.ServiceID != "" {
+		if _, err := FindServiceByID(q, filters.ServiceID); err != nil {
+			return nil, err
+		}
+		conditions = append(conditions, fmt.Sprintf("service_id = %s", q.Placeholder(idx)))
+		args = append(args, filters.ServiceID)
+		idx++
+	}
+	if filters.PMMAgentID != "" {
```

Comment:
Is there a reason for different order of those `if`s and fields in AgentFilters?